### PR TITLE
app-editors/kakoune: bump 2020.09.01

### PR DESCRIPTION
**Pull request by mujo-hash**
Signed-off-by: Ian Hixson <mujo@sdf.org>

### CODE REVIEW

**Comment by mujo-hash** [comment]
Closes: https://bugs.gentoo.org/804912

File: app-editors/kakoune/kakoune-2020.09.01.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Copyright 2020-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (juippis) [inline app-editors/kakoune/kakoune-2020.09.01.ebuild]: 8 should work here.

**Review by juippis — COMMENTED** [review]
Can you compile this on latest ~unstable? Ie, glibc-2.33, gcc-11 et all.
```
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/g++-v11/bits/stl_algo.h:3467:5: note: 'std::max' declared here
 3467 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
make: *** [Makefile:112: .regex_impl.opt.o] Error 1

...

regex_impl.cc:608:41: error: 'numeric_limits' is not a member of 'std'
  608 |         constexpr auto max_nodes = std::numeric_limits<int16_t>::max();
      |                                         ^~~~~~~~~~~~~~
regex_impl.cc:608:63: error: expected primary-expression before '>' token
  608 |         constexpr auto max_nodes = std::numeric_limits<int16_t>::max();
      |                                                               ^
regex_impl.cc:608:66: error: '::max' has not been declared; did you mean 'std::max'?
  608 |         constexpr auto max_nodes = std::numeric_limits<int16_t>::max();
      |                                                                  ^~~
      |                                                                  std::max
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/g++-v11/algorithm:62,
                 from ranges.hh:4,
                 from unicode.hh:8,
                 from utf8.hh:5,
                 from string.hh:7,
                 from exception.hh:4,
                 from regex_impl.hh:4,
                 from regex_impl.cc:1:
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/g++-v11/bits/stl_algo.h:3467:5: note: 'std::max' declared here
 3467 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
regex_impl.cc: In member function 'uint32_t Kakoune::RegexCompiler::push_inst(Kakoune::CompiledRegex::Op, uint32_t)':
regex_impl.cc:882:48: error: 'numeric_limits' is not a member of 'std'
  882 |         constexpr auto max_instructions = std::numeric_limits<int16_t>::max();
      |                                                ^~~~~~~~~~~~~~
regex_impl.cc:882:70: error: expected primary-expression before '>' token
  882 |         constexpr auto max_instructions = std::numeric_limits<int16_t>::max();
      |                                                                      ^
regex_impl.cc:882:73: error: '::max' has not been declared; did you mean 'std::max'?
  882 |         constexpr auto max_instructions = std::numeric_limits<int16_t>::max();
      |                                                                         ^~~
      |                                                                         std::max
```
To me it looks like gcc-11 related error. Is there some issue open upstream with a patch perhaps?

File: app-editors/kakoune/kakoune-2020.09.01.ebuild (hunk 1 above)
Comment (mujo-hash) [reply app-editors/kakoune/kakoune-2020.09.01.ebuild]: Done. I haven't reviewed the details of EAPI 8 so I avoided using it, thanks.

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**